command/setup_vault.go

```diff
@@ -0,0 +1,675 @@
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+
+		roleJS, _ := json.MarshalIndent(roleBody, "", "    ")
+		s.Ui.Output(string(roleJS))
+
+		var createRole bool
+		if !s.autoYes {
+			createRole = s.askQuestion(
+				"Create role in your Vault cluster? [Y/n]",
+			)
+			if !createRole {
+				s.handleNo()
+			}
+		} else {
+			createRole = true
+		}
+
+		if createRole {
+			err = s.createRoleForTasks(roleBody)
+			if err != nil {
+				s.Ui.Error(err.Error())
+				return 1
+			}
+		}
+	}
+
+	/*
+		Auth method creation
+	*/
+	s.Ui.Output(`
+Finally, we need to write a configuration for the JWT auth method.
+`)
+
+	if s.authMethodExists() {
+		s.Ui.Info("[✔] JWT Auth method configuration already exists.")
+	} else {
+
+		s.Ui.Output("This is the method configuration:\n")
+		authMethodConf, err := s.renderAuthMethod()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+		jsConf, _ := json.MarshalIndent(authMethodConf, "", "    ")
+
+		s.Ui.Output(string(jsConf))
+
+		var createAuthMethodConf bool
+		if !s.autoYes {
+			createAuthMethodConf = s.askQuestion("Create JWT auth method configuration in your Vault cluster? [Y/n]")
+			if !createAuthMethodConf {
+				s.handleNo()
+			}
+		} else {
+			createAuthMethodConf = true
+		}
+
+		if createAuthMethodConf {
+			err = s.createAuthMethod(authMethodConf)
+			if err != nil {
+				s.Ui.Error(err.Error())
+				return 1
+			}
+		}
+	}
+
+	s.Ui.Output(`
+Congratulations, your Vault cluster is now setup and ready to accept Nomad
+workloads with Workload Identity!
+
+You need to adjust your Nomad client configuration in the following way:
+
+vault {
+  enabled = true
+  address = "<Vault address>"
+
+  # Vault Enterprise only.
+  # namespace = "<namespace>"
+
+  jwt_auth_backend_path = "jwt/"
+}
+
+And your Nomad server configuration in the following way:
+
+vault {
+  enabled = true
+
+  default_identity {
+    aud = ["vault.io"]
+    ttl = "1h"
+  }
+}`)
+	return 0
+}
+
+func (s *SetupVaultCommand) jwtEnabled() bool {
+	auth, _ := s.vClient.Sys().ListAuth()
+	_, ok := auth["jwt/"]
+	return ok
+}
+
+func (s *SetupVaultCommand) enableJWT() error {
+	err := s.vClient.Sys().EnableAuthWithOptions("jwt", &api.MountInput{Type: "jwt"})
+	if err != nil {
+		return fmt.Errorf("[✘] Could not enable JWT credential backend: %w", err)
+	}
+	s.Ui.Info("[✔] Enabled JWT credential backend.")
+	return nil
+}
+
+func (s *SetupVaultCommand) roleExists() bool {
+	existingRoles, _ := s.vLogical.List("/auth/jwt/role")
+	if existingRoles != nil {
+		return slices.Contains(existingRoles.Data["keys"].([]interface{}), vaultRoleTasks)
+	}
+	return false
+}
+
+func (s *SetupVaultCommand) renderRole() (map[string]any, error) {
+	role := map[string]any{}
+	err := json.Unmarshal(vaultRoleBody, &role)
+	if err != nil {
+		return role, fmt.Errorf("[✘] Default auth config text could not be deserialized: %w", err)
+	}
+
+	role["bound_audiences"] = vaultAud
+	role["token_policies"] = []string{vaultPolicyName}
```

Comment:
We should probably not modify these and use the file asset as the source of truth. Otherwise it could get confusing to find out what needs to be changed.